data/mods/CBM_Dungeon_Test/oms_labyrinth.json

```diff
@@ -0,0 +1,88 @@
+      { "id": "vertical_dropshaft6", "into_locations": [ "subterranean_empty" ] },
+      { "id": "vertical_dropshaft7", "into_locations": [ "subterranean_empty" ] }
+    ],
+    "overmaps": {
+      "top_start_point": { "overmap": "field", "below": "vertical_dropshaft1", "locations": [ "land" ] },
+      "vert_z-1": { "overmap": "solid_earth", "above": "vertical_dropshaft1", "below": "vertical_dropshaft2" },
+      "vert_z-2": { "overmap": "deep_rock", "above": "vertical_dropshaft2", "below": "vertical_dropshaft3" },
+      "vert_z-3": { "overmap": "deep_rock", "above": "vertical_dropshaft3", "below": "vertical_dropshaft4" },
+      "vert_z-4": { "overmap": "deep_rock", "above": "vertical_dropshaft4", "below": "vertical_dropshaft5" },
+      "vert_z-5": { "overmap": "deep_rock", "above": "vertical_dropshaft5", "below": "vertical_dropshaft6" },
+      "vert_z-6": { "overmap": "deep_rock", "above": "vertical_dropshaft6", "below": "vertical_dropshaft7" },
+      "bottom_start_point": {
+        "overmap": "labyrinth_entrance_north",
+        "above": "vertical_dropshaft7",
+        "north": "passage_to_passage",
+        "east": "passage_to_passage",
+        "south": "passage_to_passage",
+        "west": "passage_to_passage"
+      },
+      "crossroads": {
+        "overmap": "labyrinth_hall_nesw",
+        "north": "passage_to_passage",
+        "east": "passage_to_passage",
+        "south": "passage_to_passage",
+        "west": "passage_to_passage"
+      },
+      "tee": {
+        "overmap": "labyrinth_hall_nes",
+        "north": "passage_to_passage",
+        "east": "passage_to_passage",
+        "south": "passage_to_passage"
+      },
+      "straight_passage": { "overmap": "labyrinth_hall_ew", "east": "passage_to_passage", "west": "passage_to_passage" },
+      "corner": { "overmap": "labyrinth_hall_ne", "north": "passage_to_passage", "east": "passage_to_passage" },
+      "dead_end": { "overmap": "labyrinth_hall_end_west", "west": "passage_to_passage" }
+    },
+    "root": "top_start_point",
+    "phases": [
+      [
+        { "overmap": "vert_z-1", "max": 1 },
+        { "overmap": "vert_z-2", "max": 1 },
+        { "overmap": "vert_z-3", "max": 1 },
+        { "overmap": "vert_z-4", "max": 1 },
+        { "overmap": "vert_z-5", "max": 1 },
+        { "overmap": "vert_z-6", "max": 1 },
+        { "overmap": "bottom_start_point", "max": 1 }
+      ],
+      [
+        { "overmap": "straight_passage", "max": { "poisson": 20 } },
+        { "overmap": "corner", "max": { "poisson": 30 } },
+        { "overmap": "tee", "max": { "poisson": 20 } },
+        { "overmap": "dead_end", "weight": 1 }
+      ],
+      [
+        { "overmap": "dead_end", "weight": 2000 },
+        { "overmap": "straight_passage", "weight": 20 },
+        { "overmap": "corner", "weight": 100 },
+        { "overmap": "tee", "weight": 20 },
+        { "overmap": "crossroads", "weight": 1 }
+      ]
+    ]
+  }
+]
```

Comment:
```suggestion
[
  {
    "type": "overmap_special",
    "id": "labyrinthine_structure",
    "subtype": "mutable",
    "locations": [ "subterranean_empty", "land" ],
    "check_for_locations": [ [ [ 0, 0, 1 ], [ "open_air" ] ], [ [ 0, 0, 0 ], [ "land" ] ] ],
    "check_for_locations_area": [ { "type": [ "subterranean_empty" ], "from": [ -4, -4, -7 ], "to": [ 4, 4, -7 ] } ],
    "//": "Placed with EoC",
    "occurrences": [ 0, 0 ],
    "flags": [ "EXODII" ],
    "joins": [ { "id": "passage_to_passage", "into_locations": [ "subterranean_empty" ] }, { "id": "surface_to_z-7" } ],
    "overmaps": {
      "root": { "overmap": "field", "above": "surface_to_z-7" },
      "root_to_entrance": { "overmap": "open_air", "below": "surface_to_z-7" },
      "entrance": {
        "overmap": "labyrinth_entrance_north",
        "north": "passage_to_passage",
        "east": "passage_to_passage",
        "south": "passage_to_passage",
        "west": "passage_to_passage"
      },
      "crossroads": {
        "overmap": "labyrinth_hall_nesw",
        "north": "passage_to_passage",
        "east": "passage_to_passage",
        "south": "passage_to_passage",
        "west": "passage_to_passage"
      },
      "tee": {
        "overmap": "labyrinth_hall_nes",
        "north": "passage_to_passage",
        "east": "passage_to_passage",
        "south": "passage_to_passage"
      },
      "straight_passage": { "overmap": "labyrinth_hall_ew", "east": "passage_to_passage", "west": "passage_to_passage" },
      "corner": { "overmap": "labyrinth_hall_ne", "north": "passage_to_passage", "east": "passage_to_passage" },
      "dead_end": { "overmap": "labyrinth_hall_end_west", "west": "passage_to_passage" }
    },
    "root": "root",
    "phases": [
      [
        {
          "name": "entrance",
          "chunk": [ { "overmap": "root_to_entrance", "pos": [ 0, 0, 1 ] }, { "overmap": "entrance", "pos": [ 0, 0, -7 ] } ],
          "max": 1
        }
      ],
      [
        { "overmap": "straight_passage", "max": { "poisson": 20 } },
        { "overmap": "corner", "max": { "poisson": 30 } },
        { "overmap": "tee", "max": { "poisson": 20 } },
        { "overmap": "dead_end", "weight": 1 }
      ],
      [
        { "overmap": "dead_end", "weight": 2000 },
        { "overmap": "straight_passage", "weight": 20 },
        { "overmap": "corner", "weight": 100 },
        { "overmap": "tee", "weight": 20 },
        { "overmap": "crossroads", "weight": 1 }
      ]
    ]
  }
]

```
Replaced the dropshaft with a single z1 map instead so it doesn't care what's at z-1 through z-6
Made it do an area check for subterranean_empty so it doesn't gen too tiny, happily change the size or omit this though if it needs to be able to be placed near overmap edges etc
I can't see where you spawn the labyrinth but presuming the [ 0, 0 ] occurrences is intentional adding a comment is probably a good idea (you could specify which EoC/dialogue or whatever does it)
I haven't tested in game, the open_air map might need "locations": "open_air" but I'm pretty sure it defaults to that because that's the base region setting terrain for z1, other than that it's linted and I can't think of any other obvious potential issues

Reply:
I don't quite understand what you're doing here, but I will give it a shot and curse your name if it fails.